app/tip/[walletAddress]/page.tsx

```diff
@@ -126,6 +129,28 @@ export default function Tip() {
     }, 1000);
   };
 
+  const handleEip712TapToPay = useCallback(async () => {
+    // make a POST request to the NFC relayer
+    const response = await fetch(process.env.NEXT_PUBLIC_NFC_RELAYER_URL as string, {
```

Comment:
nit: let's wrap `process.env` variables into a global config object that casts to the correct type.